pkgs/native_toolchain_c/test/helpers.dart

```diff
@@ -121,30 +120,31 @@ extension on Uri {
 String unparseKey(String key) => key.replaceAll('.', '__').toUpperCase();
 
 /// Archiver provided by the environment.
-final Uri? ar = Platform
-    .environment[unparseKey(CCompilerConfig.arConfigKeyFull)]
-    ?.asFileUri();
+///
+/// Provided on Dart CI.
+final Uri? ar = Platform.environment[unparseKey('c_compiler.ar')]?.asFileUri();
```

Comment:
I don't know what is happening here, so rubberstamp-LGTMing :)

Reply:
The tests run on the Dart CI as well in the `pkg/` bots.
On the Dart CI, we provide the compilers that are already installed by the Dart SDK checkout instead of redownloading them every build. (On the GitHub bots we redownload clang every single run...) This environment key gives the path to the compiler archiver (`ar`).